test/performance/search/dream_index/interleaved_bloom_filter_benchmark.cpp

```diff
@@ -20,13 +21,19 @@ inline benchmark::Counter hashes_per_second(size_t const count)
 
 static void arguments(benchmark::internal::Benchmark* b)
 {
+    // Bins must be powers of two
     for (int32_t bins : {64, 8192})
     {
-        for (int32_t bits = 1<<15; bits <= 1<<20/* Increase for more extensive benchmarks*/; bits <<= 5)
+        // Size of the IBF will be 2^bits bits
+        for (int32_t bits = 15; bits <= 20; bits += 5)
         {
-            for (int32_t hash_num = 2; hash_num < 3/* Increase for more extensive benchmarks*/; ++hash_num)
+            // The bits per bin must fit in an int32_t
+            if (bits - std::countr_zero(static_cast<uint32_t>(bins)) < 32)
             {
-                b->Args({bins, bits/bins, hash_num, 1'000/* Increase for more extensive benchmarks*/});
+                for (int32_t hash_num = 2; hash_num < 3; ++hash_num)
+                {
+                    b->Args({bins, (1LL << bits)/bins, hash_num, 1'000});
```

Comment:
Why did you add the 1LL here? And why did you remove the comment to increase 1000 for more extensive benchmarks?

Reply:
Before, `bits` was the overall IBF size; but googlebenchmark only accepts `int32_t` as input for arguments. So we were limited to at most a size of 2^31 basically.
What was worse was that if you put `bits <= 35` (or anything > 31), there was an overflow and the program did not terminate and consumed all memory.

Now you set the exponent, i.e. `2^bits`. `1LL << bits` is equivalent to `2^bits`. We use signed integers to not need to cast again. 
And we can also avoid the problem with `bits > 31`.

I removed the comments because it's basically the same in every benchmark we have: if you want to benchmark more, you need to change parameters.